drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/migration/mci_mapping_importer.rb

```diff
@@ -83,9 +85,12 @@ def lookup
       parsed = sheet.parse(headers: true)
 
       parsed.each do |row|
+        mci_uniq_id = row['MCI_UNIQ_ID']&.to_i&.to_s
         mci_id = row['MCI_ID']&.to_i&.to_s
+        next unless mci_uniq_id.present? && mci_id.present?
 
-        @lookup[row['MCI_UNIQ_ID']&.to_i&.to_s] = mci_id if mci_id.present?
+        @lookup[mci_uniq_id] = [] unless @lookup.key?(mci_uniq_id)
+        @lookup[mci_uniq_id] << mci_id if mci_id.present?
```

Comment:
The bug was that the job only expected one mci id per mci unique id. In fact, the mci unique ids can be mapped to multiple mci ids, and we need to create all of them. I realized this when seeing lots of unrecognized mci ids when importing referrals.